src/components/fields/wrapper/wrapper.tsx

```diff
@@ -0,0 +1,77 @@
+import isEmpty from "lodash/isEmpty";
+import React, { useEffect, useState } from "react";
+import { Controller, useFormContext } from "react-hook-form";
+import { EFieldType, IGenericFieldProps, TFrontendEngineFieldSchema } from "src/components/frontend-engine/types";
+import * as FrontendEngineFields from "../";
+import { IWrapperSchema } from "./types";
+
+interface IWrapperProps {
+	id?: string;
+	schema?: IWrapperSchema | undefined;
+	/** only used internally by FrontendEngine */
+	children?: Record<string, TFrontendEngineFieldSchema>;
+}
+
+export const Wrapper = (props: IWrapperProps): JSX.Element | null => {
+	// =============================================================================
+	// CONST, STATE, REF
+	// =============================================================================
+	const { id, schema, children } = props;
+	const { fieldType, children: schemaChildren, ...otherSchema } = schema || {};
+	const [fields, setFields] = useState<React.ReactNode>(null);
+	const { control } = useFormContext();
+
+	// =============================================================================
+	// EFFECTS
+	// =============================================================================
+	useEffect(() => {
```

Comment:
note: this is mostly the same code moved from `FrontendEngine`. essentially the wrapper component is rendering the fields the same way as `FrontendEngine` hence the code is moved here to avoid repeating the same code